src/main/webapp/app/iris/exercise-chatbot/exercise-chatbot-button.component.ts

```diff
@@ -1,31 +1,64 @@
-import { Component, Input, OnDestroy, OnInit } from '@angular/core';
+import { Component, ElementRef, Input, OnDestroy, OnInit, ViewChild } from '@angular/core';
 import { MatDialog, MatDialogRef } from '@angular/material/dialog';
 import { Overlay } from '@angular/cdk/overlay';
 import { ActivatedRoute } from '@angular/router';
 import { IrisChatbotWidgetComponent } from 'app/iris/exercise-chatbot/widget/chatbot-widget.component';
-import { Subscription } from 'rxjs';
-import { faChevronDown, faCircle } from '@fortawesome/free-solid-svg-icons';
+import { EMPTY, Subscription, filter, of, switchMap } from 'rxjs';
+import { faAngleDoubleDown, faChevronDown, faCircle } from '@fortawesome/free-solid-svg-icons';
 import { IrisLogoLookDirection, IrisLogoSize } from 'app/iris/iris-logo/iris-logo.component';
 import { ChatServiceMode, IrisChatService } from 'app/iris/iris-chat.service';
+import { animate, state, style, transition, trigger } from '@angular/animations';
+import { IrisTextMessageContent } from 'app/entities/iris/iris-content-type.model';
 
 @Component({
     selector: 'jhi-exercise-chatbot-button',
     templateUrl: './exercise-chatbot-button.component.html',
     styleUrls: ['./exercise-chatbot-button.component.scss'],
+    animations: [
+        trigger('expandAnimation', [
+            state(
+                'hidden',
+                style({
+                    opacity: 0,
+                    transform: 'scale(0)',
+                    transformOrigin: 'bottom right',
+                }),
+            ),
+            state(
+                'visible',
+                style({
+                    opacity: 1,
+                    transform: 'scale(1)',
+                    transformOrigin: 'bottom right',
+                }),
+            ),
+            transition('hidden => visible', animate('300ms ease-out')),
+            transition('visible => hidden', animate('300ms ease-in')),
+        ]),
+    ],
 })
 export class IrisExerciseChatbotButtonComponent implements OnInit, OnDestroy {
     @Input()
     mode: ChatServiceMode;
 
     dialogRef: MatDialogRef<IrisChatbotWidgetComponent> | null = null;
     chatOpen = false;
+    isOverflowing = false;
     hasNewMessages = false;
+    newIrisMessage: string | undefined;
+
+    private readonly CHAT_BUBBLE_TIMEOUT = 10000;
+
     private numNewMessagesSubscription: Subscription;
     private paramsSubscription: Subscription;
+    private latestIrisMessageSubscription: Subscription;
 
     // Icons
     faCircle = faCircle;
     faChevronDown = faChevronDown;
+    faAngleDoubleDown = faAngleDoubleDown;
+
+    @ViewChild('chatBubble') chatBubble: ElementRef;
 
```

Comment:
Can you adapt the file to use the new client guidelines? With inject instead  of the constructor, and using signals?

Reply:
There is currently a separate PR for migrating Iris components according to latest coding guidelines for angular.

Please see #9615